clang/include/clang/Basic/DiagnosticGroups.td

```diff
@@ -1547,6 +1547,9 @@ def DXILValidation : DiagGroup<"dxil-validation">;
 // Warning for HLSL API availability
 def HLSLAvailability : DiagGroup<"hlsl-availability">;
 
+// Warnings for legacy binding behavior
+def DisallowLegacyBindingRules : DiagGroup<"disallow-legacy-binding-rules">;
+
```

Comment:
I think it would be better to name the group `legacy-binding-rules`. The verb `disallow` just makes it confusing when turning these into warnings like this: `-Wno-error=legacy-binding-rules`.

Reply:
This reminds me, the spec actually specified what the group name should be: `LegacyConstantRegisterBinding`
I'll change it to that so that it matches the spec.